experimental/cache_size.cc

```diff
@@ -0,0 +1,65 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include <algorithm>
+#include <iostream>
+
+int64_t MAX_SIZE = 16 * 1024 * 1024;
+int64_t MIN_SIZE = 1024;
+int64_t ACCESSES_NUM = 64 * 1024 * 1024;
+int REPEATS = 50;
+
+clock_t whack_cache(const int64_t sz, int64_t* accesses) {
+  char* buf = new char[sz];
+
+  clock_t start = clock();
+  // TODO fencing
```

Comment:
Ah, the timing is O(n). I think this explains the positively sloped caching behavior, whereas for each level of cache it should ideally be flat -- if we made it O(1) e.g. by measuring the time it takes to read only the first written cache line, it might even be easier to tease out the distinction between the different cache levels.

Do you agree / want to add a TODO here to make the timing O(1)?

(This also might help with e.g. measuring cache line size.)

Reply:
I am not sure which part of the code you are referring here.